### PR TITLE
docs: add terminology page to explain Trivy concepts

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
This PR adds a new terminology page that documents the key concepts and terms used in Trivy. 
The page covers core components, scanning features, and infrastructure elements.

Note: The term "Scan Assets" is newly introduced in this PR to collectively refer to external data that Trivy downloads and uses during scanning (e.g., vulnerability database, Java database, checks bundle).

The goal is to provide a single source of truth for Trivy terminology and help users better understand the project's architecture and components.

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
@aquasecurity/trivy The definition of the terminology system will be helpful for our communication and document updates, so please look over it and feel free to leave any feedback.

File: docs/docs/references/terminology.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+- [Vulnerability Scanner](../scanner/vulnerability.md): Detects known vulnerabilities in OS packages and application dependencies
+- [Misconfiguration Scanner](../scanner/misconfiguration/index.md): Identifies security misconfigurations in IaC files
+- [Secret Scanner](../scanner/secret.md): Finds hardcoded secrets and sensitive information
+- [License Scanner](../scanner/license.md): Identifies software license issues
+
+### Scan Assets
+External data that Trivy downloads and uses during scanning:
+
+- [Vulnerability Database (Trivy DB, trivy-db)](#vulnerability-database-trivy-db-trivy-db)   : Database containing vulnerability information
```
Comment (simar7) [inline docs/docs/references/terminology.md:21]:
```suggestion
- [Vulnerability Database (Trivy DB, trivy-db)](#vulnerability-database-trivy-db-trivy-db): Database containing vulnerability information
```

File: docs/docs/references/terminology.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+A GitHub repository that collects and stores vulnerability information from various data sources.
+This repository serves as the foundation for building the Trivy DB.
+
+Managed at:
+
+- https://github.com/aquasecurity/vuln-list
+- https://github.com/aquasecurity/vuln-list-nvd
+- https://github.com/aquasecurity/vuln-list-redhat
+- https://github.com/aquasecurity/vuln-list-debian
+- etc.
```
Comment (simar7) [inline docs/docs/references/terminology.md:51]: Can you make them as hyperlinks?

File: docs/docs/references/terminology.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+A threshold value that determines the reliability of license detection.
+Higher values increase detection accuracy but may miss some licenses, while lower values catch more potential licenses but may include false positives.
+
+## Kubernetes Integration
+
+### Trivy Operator
+Kubernetes operator for Trivy that enables continuous security scanning in Kubernetes clusters.
+The operator automates vulnerability and misconfiguration scanning of workloads running in a cluster.
+
+Managed at https://github.com/aquasecurity/trivy-operator.
```
Comment (simar7) [inline docs/docs/references/terminology.md:104]: ditto hyperlink

File: docs/docs/references/terminology.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+### Plugin
+An add-on tool that integrates with Trivy to extend its core functionality.
+Plugins can be written in any programming language and integrate seamlessly with Trivy CLI, appearing in Trivy help and subcommands.
+They can be installed and removed independently without affecting the core Trivy installation.
+
+For detailed information about plugins, see [the document](../plugin/index.md).
+
+### Plugin Index
+A centralized registry that lists available Trivy plugins, managed at https://github.com/aquasecurity/trivy-plugin-index.
```
Comment (simar7) [inline docs/docs/references/terminology.md:159]: ditto hyperlink

File: docs/docs/references/terminology.md (hunk 2 above)
Comment (knqyf263) [reply docs/docs/references/terminology.md:66]: done https://github.com/aquasecurity/trivy/pull/7996/commits/306d6b021420c49b1774de525651ce0f58245808

File: docs/docs/references/terminology.md (hunk 3 above)
Comment (knqyf263) [reply docs/docs/references/terminology.md]: done https://github.com/aquasecurity/trivy/pull/7996/commits/306d6b021420c49b1774de525651ce0f58245808

**Review by simar7 — APPROVED** [review]

**Comment by simar7** [comment]
LGTM, but I'll wait for @nikpivkin and @DmitriyLewen to also take a look.

**Review by nikpivkin — APPROVED** [review]

File: docs/docs/references/terminology.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+### Scanner
+Trivy's built-in security scanning engines. Trivy has four main scanners:
+
+- [Vulnerability Scanner](../scanner/vulnerability.md)
+- [Misconfiguration Scanner](../scanner/misconfiguration/index.md)
+- [Secret Scanner](../scanner/secret.md)
+- [License Scanner](../scanner/license.md)
+
+### Scan Assets
+External data that Trivy downloads and uses during scanning:
```
Comment (DmitriyLewen) [inline docs/docs/references/terminology.md:41]: But I'm not sure that's required here:
```suggestion
External data that Trivy downloads (if needed for scanner) and uses during scanning:
```

File: docs/docs/references/terminology.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    - Usage instructions (these belong in feature documentation)
+
+
+## Core Concepts
+
+### Target
+Types of artifacts that Trivy can scan, like container images and filesystem.
+
+### Scanner
+Trivy's built-in security scanning engines. Trivy has four main scanners:
```
Comment (DmitriyLewen) [inline docs/docs/references/terminology.md:33]: The `scanners` flag only allows these 4 scanners.
But there is "shadow" scanner `sbom` (e.g. when we use the SBOM format).

Is it worth mentioning?

File: docs/docs/references/terminology.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+## Vulnerability Scanning
+
+### Vulnerability Database (Trivy DB, trivy-db)
+The core vulnerability database required for vulnerability detection.
+Contains comprehensive vulnerability information for multiple ecosystems.
+Distributed via OCI registry.
+
+Managed at https://github.com/aquasecurity/trivy-db.
+
+The vulnerability database is built from A GitHub repository that collects and stores vulnerability information from various data sources.
```
Comment (DmitriyLewen) [inline docs/docs/references/terminology.md:57]: nit:
```suggestion
The vulnerability database is built from a GitHub repository that collects and stores vulnerability information from various data sources.
```

File: docs/docs/references/terminology.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+This repository serves as the foundation for building the Trivy DB.
+
+Managed at:
+
+- https://github.com/aquasecurity/vuln-list
+- https://github.com/aquasecurity/vuln-list-nvd
+- https://github.com/aquasecurity/vuln-list-redhat
+- https://github.com/aquasecurity/vuln-list-debian
+- etc.
+-
```
Comment (DmitriyLewen) [inline docs/docs/references/terminology.md:67]: This `-` creates extra dot:

Comment (knqyf263) [reply docs/docs/references/terminology.md]: Thanks. Deleted https://github.com/aquasecurity/trivy/pull/7996/commits/26d3cf40d5a0f51a43d0e969431d4a277155acd3

File: docs/docs/references/terminology.md (hunk 6 above)
Comment (knqyf263) [reply docs/docs/references/terminology.md:33]: Makes sense. Mentioned.
https://github.com/aquasecurity/trivy/pull/7996/commits/6e8cf5f80132aaf1d6f19b74740ee3247ef68203

**Review by DmitriyLewen — APPROVED** [review]

**Comment by DmitriyLewen** [comment]
@knqyf263 can you rewrite CLA?

**Comment by knqyf263** [comment]
Tip: You can request that the CLA assistant run again by https://cla-assistant.io/check/aquasecurity/trivy?pullRequest=7996.

**Pull request merged**